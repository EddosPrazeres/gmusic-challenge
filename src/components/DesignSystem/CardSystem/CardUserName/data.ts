export interface ICardUserName {   
  avatar_url: string;
  login: string;
  link?: string;
}

export interface ICardUserNameLayout extends ICardUserName { 
  avatar_url: string;
  login: string;
}