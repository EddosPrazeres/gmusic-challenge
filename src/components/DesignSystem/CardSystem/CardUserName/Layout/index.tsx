import * as React from "react";
import { CardWrapper, Text } from "../../../../../components";
import { ICardUserNameLayout } from "../data";
import './styles.scss'

export function CardUserName({ avatar_url, login, link}: ICardUserNameLayout) {
  const id = login;
  return (
    <CardWrapper {...link && {link: `${link}/${id}`}}> 
        <div className="CardUserName">
          <img className="CardUserName-avatar" src={avatar_url}/>
          <Text title={login} size="large" />          
        </div>
    </CardWrapper>
  )
}