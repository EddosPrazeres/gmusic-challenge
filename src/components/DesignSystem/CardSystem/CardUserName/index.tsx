import * as React from 'react';
import { ICardUserName } from './data';
import { CardUserName as Layout } from './Layout';

export default function CardUserName(props: ICardUserName){
  return (
    <Layout {...props}/>
  )
}

  