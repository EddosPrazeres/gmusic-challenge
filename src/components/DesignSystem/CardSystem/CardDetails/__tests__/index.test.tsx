import React from 'react'
import { mount, shallow } from 'enzyme'
import CardUserName from '../index'

const createComponent = (props, useDefault = false) => {
  const componentProps = {    
    ...props,
  }

  const wrapper = useDefault ? mount(<CardUserName  {...componentProps} />) : shallow(<CardUserName     {...componentProps} />)
  const unmount = () => wrapper.unmount()

  return [wrapper, unmount]
}

describe('CardUserName', () => {
  it('should match snapshot', () => {
    const [wrapper, unmount] = createComponent({})

    expect(wrapper).toMatchSnapshot()
    unmount()
  })
})
