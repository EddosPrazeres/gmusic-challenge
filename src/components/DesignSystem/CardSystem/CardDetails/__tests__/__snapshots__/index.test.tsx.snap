// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardUserName should match snapshot 1`] = `ShallowWrapper {}`;
